### ISSUE TITLE
[BUG] neg_log_loss scorer incompatible with newer scikit-learn version

### ISSUE THREAD

**Issue by cam-rogers** [deepchecks/deepchecks#2806]
**Describe the bug**
When trying to run the deepchecks TrainTestPerformance check with 'neg_log_loss' as a custom scorer, I run into an issue with how the code builds the log loss scorer using make_scorer. It tries to define the argument needs_proba=True in the make_scorer function. In an updated scikit-learn version (1.7.2), that argument is deprecated. Here is the location of the error: 

https://github.com/deepchecks/deepchecks/blob/main/deepchecks/tabular/metric_utils/scorers.py#L151

If we instead called get_scorer("neg_log_loss"), the following line would be the up to date way to define the log loss scorer via scikit-learn:

make_scorer(
    log_loss, greater_is_better=False, response_method="predict_proba"
)

**To Reproduce**
Here is some sample code to reproduce the behavior (if on scikit-learn version 1.7.2 or above):

```
from deepchecks.tabular import Dataset
from deepchecks.tabular.checks import TrainTestPerformance
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
import json

# Load iris data
iris = load_iris(as_frame=True)
X = iris.data
y = iris.target

# Train/test split
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# Fit logistic regression
clf = LogisticRegression(max_iter=200)
clf.fit(X_train, y_train)

# Predict probabilities
probs = clf.predict_proba(X_test)
probs_train = clf.predict_proba(X_train)

# Create Deepchecks Datasets
train_ds = Dataset(
    X_train,
    label=y_train,
    features=list(X.columns),
    label_type="multiclass",
    cat_features=[],
)
test_ds = Dataset(
    X_test,
    label=y_test,
    features=list(X.columns),
    label_type="multiclass",
    cat_features=[],
)

# Run TrainTestPerformance check
check = TrainTestPerformance(scorers=["neg_log_loss"])
result = check.run(
    train_ds,
    test_ds,
    y_proba_test=probs,
    y_proba_train=probs_train,
    model_classes=[0, 1, 2],
)
```

**Environment (please complete the following information):**
 - OS: Windows 11
 - Python Version: 3.12.11
 - Deepchecks Version: 0.19.1

Let me know if this can be patched, thank you!
